### PR TITLE
fix(integration-karma): run hydration preprocessor just for focused tests

### PR DESCRIPTION
**Pull request by cardoso**
## Details

I found it a bit difficult to debug the server-side code when using the .only as the transformation will run for all tests.

I added this quick solution to run the transformation just for the focused tests (probably can be improved).

## Does this pull request introduce a breaking change?

-   😮‍💨 No, it does not introduce a breaking change.

## Does this pull request introduce an observable change?

-   🤞 No, it does not introduce an observable change.

### CODE REVIEW

File: packages/@lwc/integration-karma/scripts/karma-configs/hydration/base.js [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
 const path = require('path');
 
+const { globSync } = require('fs');
```
Comment (nolanlawson) [inline packages/@lwc/integration-karma/scripts/karma-configs/hydration/base.js:12]:
```suggestion
const { globSync } = require('node:fs');
```

We're not avoiding `node:*` intentionally; we just haven't fixed every missing `node:` prefix. Same for the `path` above – feel free to clean up as you go. 🙂

File: packages/@lwc/integration-karma/scripts/karma-configs/hydration/base.js [hunk 2]
```diff
@@ -36,14 +37,27 @@ const ALL_FRAMEWORK_FILES = [SYNTHETIC_SHADOW, LWC_ENGINE];
...
         createPattern(TEST_UTILS),
         createPattern(TEST_HYDRATE),
-        createPattern('**/*.spec.js', { watched: false }),
     ];
+
+    // check if a .only file exists
+    const onlyFile = globSync('**/*/.only', { cwd: BASE_DIR, absolute: true });
+
+    if (onlyFile.length > 1) {
+        throw new Error(`More than one .only file found in ${BASE_DIR}`);
```
Comment (nolanlawson) [inline packages/@lwc/integration-karma/scripts/karma-configs/hydration/base.js:52]: Sometimes we want to test two things at once... I wouldn't throw here. I would just push multiple patterns if `onlyFile.length > 0`.

**Review by nolanlawson — CHANGES_REQUESTED** [review]
LGTM but a few small nits. Thank you!

File: packages/@lwc/integration-karma/scripts/karma-configs/hydration/base.js (hunk 2 above)
Comment (cardoso) [reply packages/@lwc/integration-karma/scripts/karma-configs/hydration/base.js]: Makes sense! I made the change and it seems to work well.

File: packages/@lwc/integration-karma/scripts/karma-configs/hydration/base.js (hunk 1 above)
Comment (cardoso) [reply packages/@lwc/integration-karma/scripts/karma-configs/hydration/base.js]: Changed that one and the path require too.

**Review by nolanlawson — APPROVED** [review]

**Comment by nolanlawson** [comment]
Thanks!

**Comment by nolanlawson** [comment]
Somehow this merged via GitHub auto-merge despite failing the `run-karma-tests-group-4` test due to Node v20 not having [globSync](https://nodejs.org/api/fs.html#fsglobsyncpattern-options).

I'll have to fix our CI settings, but I'll open up another PR to fix the glob issue...

**Pull request merged**